18_regexp_lab/03.py

```diff
@@ -1,16 +1,12 @@
-"""
-Write a python program that takes a CSV file
-reads it line by line and prints each line
-with first and second columns reversed.
-
-Sample input:
-    Shana,Sargent,shanasargent@isoswitch.com
-    Witt,Hampton,witthampton@zaphire.com
-    Morgan,Grant,morgangrant@lotron.com
-
-Sample output:
-    Sargent,Shana,shanasargent@isoswitch.com
-    Hampton,Witt,witthampton@zaphire.com
-    Grant,Morgan,morgangrant@lotron.com
-"""
+#!/usr/bin/python
+import re 
+import sys, os
 
+with open('csv', 'r') as f:
+	for line in f:
+		x = re.search(r',\b(\w+)', line)
+		y = re.search(r'\b(\w+)' ,line)
+		if x and y:
+			line =  line.replace(y.group(1) , '!')
+			line =  line.replace(x.group(1) , y.group(1))
+			print line.replace( '!', x.group(1))
```

Comment:
But what if a column contains `!` as part of its text?

I think it'd be better to use split here, and assign correct values by indexes.
Better yet, read about CSV module here and use it:
https://docs.python.org/2/library/csv.html